libs/store-selection/src/lib/store-selection.tsx

```diff
@@ -1,23 +1,132 @@
-import { useEffect } from 'react';
-import styled from 'styled-components';
+import { useEffect, useState } from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
 
 // eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
 import { getData } from '../../../../apps/online-store/src/utils/APIrequest';
+import { FilterMenus } from './filter-menus/filter-menus';
+import { PageButtons } from './page-buttons/page-buttons';
+import { StoreCard } from './store-card/store-card';
 /* eslint-disable-next-line */
 export interface StoreSelectionProps {}
 
-const StyledStoreSelection = styled.div`
-  color: red;
-`;
-
 export function StoreSelection(props: StoreSelectionProps) {
+  const [storesAndMenus, setStoresAndMenus] = useState<any>([]);
+  const [pages, setPages] = useState<number>(0);
+  const [storeName, setStoreName] = useState<string>('');
+  const [activeBtn, setActiveBtn] = useState<string>('');
+  const { page, deliveryType, nameOfStore } = useParams();
+  const navigate = useNavigate();
+  const location = useLocation();
+  let link;
+  let newLink;
+
+  const handleStoreInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setStoreName(event.target.value);
+    if (page && nameOfStore) {
+      link = location.pathname.split('/');
+      link.pop();
+      newLink = link.join('/');
+      navigate(`${newLink}/${event.target.value}`);
+    } else if (page && deliveryType) {
+      link = location.pathname.split('/');
+      link.pop();
+      newLink = link[0];
+      navigate(`${newLink}/store-selection/1/${event.target.value}`);
+    } else if (page) {
+      link = location.pathname.split('/');
+      link.pop();
+      newLink = link.join('/');
+      navigate(`${newLink}/1/${event.target.value}`);
+    } else {
+      navigate(`1/${event.target.value}`);
+    }
+  };
+
   useEffect(() => {
```

Comment:
pls, add one more condition to check the URL when it includes store name and page, because in my opinion there could be a bug.

Reply:
Done